websocket.el

```diff
@@ -693,14 +693,27 @@ to the websocket protocol.
                           (port (if (= 0 (url-port url-struct))
                                     (if (eq type 'tls) 443 80)
                                   (url-port url-struct)))
-                          (host (url-host url-struct)))
+                          (host (url-host url-struct))
+                          (url-as-http (let ((url-as-http (copy-sequence url-struct)))
+                                         (setf (url-type url-as-http) (if (eq type 'plain) "http" "https"))
+                                         url-as-http))
+                          (proxy (url-generic-parse-url (url-find-proxy-for-url url-as-http (url-host url-as-http)))))
                      (if (eq type 'plain)
                          (make-network-process :name name :buffer nil :host host
                                                :service port :nowait nowait)
+                       (if proxy
+                           (let ((plain-conn (make-network-process
+                                              :name name :buffer nil :host (url-host proxy)
+                                              :service (url-port proxy))))
+                             (let ((url-http-after-change-function)
+                                   (url-current-object url-as-http))
+                               (url-https-proxy-connect plain-conn))
+                             (sleep-for 0.5)
```

Comment:
Why is the sleep needed?  Can we instead check `plain-conn`?

Reply:
We need to wait until we get a response to the CONNECT issued by url-https-proxy-connect. The proper (and only) way to do this is with a process-filter. I was just too lazy to hook up that additional possible state into the websocket.el process filter...